src/platform/packages/private/kbn-grid-layout/grid/use_grid_layout_events/sensors/autoscroll.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { isMouseEvent } from '.';
+
+const MIN_SPEED = 50;
+const MAX_SPEED = 150;
+
+let scrollInterval: NodeJS.Timeout | null = null;
+
+const scrollOnInterval = (direction: 'up' | 'down') => {
+  let count = 0;
+  let currentSpeed = MIN_SPEED;
+  let maxSpeed = MIN_SPEED;
+  let turnAroundPoint: number | undefined;
+
+  scrollInterval = setInterval(() => {
+    /**
+     * Since "smooth" scrolling on an interval is jittery on Chrome, we are manually creating
+     * an "ease" effect via the parabola formula `y = a(x - h)^2 + k`
+     *
+     * Scrolling slowly speeds up as the user drags, and it slows down again as they approach the
+     * top and/or bottom of the screen.
+     */
+    const nearTop = direction === 'up' && scrollY < window.innerHeight;
+    const nearBottom =
+      direction === 'down' &&
+      window.innerHeight + window.scrollY > document.body.scrollHeight - window.innerHeight;
+    if (!turnAroundPoint && (nearTop || nearBottom)) {
+      // reverse the direction of the parabola
+      maxSpeed = currentSpeed;
+      turnAroundPoint = count;
+    }
+
+    currentSpeed = turnAroundPoint
+      ? Math.max(-3 * (count - turnAroundPoint) ** 2 + maxSpeed, MIN_SPEED) // slow down fast
+      : Math.min(0.1 * count ** 2 + MIN_SPEED, MAX_SPEED); // speed up slowly
+    window.scrollBy({
+      top: direction === 'down' ? currentSpeed : -currentSpeed,
+    });
+
+    count++; // increase the counter to increase the time interval used in the parabola formula
+  }, 60);
+  return scrollInterval;
+};
+
+export const stopAutoScroll = () => {
+  if (scrollInterval) {
+    clearInterval(scrollInterval);
+    scrollInterval = null;
+  }
+};
+
+export const handleAutoscroll = (e: Event) => {
+  if (!isMouseEvent(e)) return;
```

Comment:
Do we still need this check? Since we are only calling `handleAutoScroll` in `startMouseInteraction`

Reply:
That was here because we only call it there, but we also use it for scroll event (which we don't want to do). But I think you're right - instead I made some changes in types and how I call this function f031a27074650bbb45b9fded5b282d971ce8c5fd